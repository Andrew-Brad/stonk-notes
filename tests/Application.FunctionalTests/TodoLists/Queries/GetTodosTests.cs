using Snapshooter.NUnit;
using StonkNotes.Application.TodoLists.Queries.GetTodos;
using StonkNotes.Domain.Entities;
using StonkNotes.Domain.ValueObjects;

namespace StonkNotes.Application.FunctionalTests.TodoLists.Queries;

using static Testing;

public class GetTodosTests : BaseTestFixture
{
    [Test]
    public async Task ShouldReturnPriorityLevels()
    {
        await RunAsDefaultUserAsync();

        var query = new GetTodosQuery();

        var result = await SendAsync(query);

        result.PriorityLevels.Should().NotBeEmpty();
    }

    [Test]
    public async Task ShouldReturnAllListsAndItems()
    {
        await RunAsDefaultUserAsync();

        await AddAsync(new TodoList
        {
            Title = "Shopping",
            Colour = Colour.Blue,
            Items =
                    {
                        new TodoItem { Title = "Apples", Done = true },
                        new TodoItem { Title = "Milk", Done = true },
                        new TodoItem { Title = "Bread", Done = true },
                        new TodoItem { Title = "Toilet paper" },
                        new TodoItem { Title = "Pasta" },
                        new TodoItem { Title = "Tissues" },
                        new TodoItem { Title = "Tuna" }
                    }
        });

        var query = new GetTodosQuery();

        var result = await SendAsync(query);

        result.Lists.Should().HaveCount(1);
        result.Lists.First().Items.Should().HaveCount(7);
    }

    [Test]
    public async Task ShouldDenyAnonymousUser()
    {
        var query = new GetTodosQuery();

        var action = () => SendAsync(query);

        await action.Should().ThrowAsync<UnauthorizedAccessException>();
    }

    // testing only to ensure the simplest possible thing works
    [Test]
    public async Task CreateSampleSnapshotTest()
    {
        // arrange
        await Task.CompletedTask;
        //var serviceClient = new ServiceClient();

        // act
        ShortText text = ShortText.From(Guid.Parse("2292F21C-8501-4771-A070-C79C7C7EF451").ToString());

        // assert
        Snapshot.Match(text);
    }
}
