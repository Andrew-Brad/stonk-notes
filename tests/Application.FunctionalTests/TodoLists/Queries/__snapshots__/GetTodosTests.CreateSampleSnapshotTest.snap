{
  "TextBody": "2292f21c-8501-4771-a070-c79c7c7ef451"
}
